src/wwwroot/app/views/candidate/candidate.controller.js

```diff
@@ -1,14 +1,22 @@
-export default function CandidateController($scope, CandidateService) {
+export default function CandidateController($scope, CandidateService, $validation) {
    'ngInject';
 
    var vm = $scope;
-   vm.saveCandidate = saveCandidate;
-
-   function saveCandidate() {
-      CandidateService.saveCandidate(vm.candidate).catch(_onError);
-   }
+   vm.submit = _submit;
+   vm.reset = _reset;
 
    function _onError(message) {
       vm.errorMessage = 'Sorry! Some error occurred';
    }
+
+   function _submit(form) {
+      $validation.validate(form);
+      if ($validation.checkValid(form)) {
```

Comment:
All of controllers will be contain this two lines
so, maybe, we need create some service for it.
and incapsulate all work with validation inside

And our controllers will be much better, something like this

```
ValidationService.validate(form).then(() => CandidateService.saveCandidate(vm.candidate).catch(_onError))
```